packages/docusaurus/src/server/html-tags/__tests__/__fixtures__/plugin-both-tags.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+module.exports = function() {
+  return {
+    name: 'plugin-both-tags',
+    injectHtmlTags() {
+      return {
+        headTags: [
```

Comment:
What do you think of allowing an additional form of injecting tags using strings? It's a bit messy but might be more ergonomic to use.

```
headTags: [
  { tagName: 'link', ... },
  '<link rel="preconnect" href="www.google-analytics.com"/>',
],
```

Reply:
I'd prefer html tag object because we can validate if its valid html tag. 

Doing it via string can be dangerous, we need to parse the string if its a valid tagname, closing is correct, checking if its a `html void element`, etc.
We can ignore it but I'd prefer sticking with https://github.com/jantimon/html-webpack-plugin API like above

Reply:
We don't need to parse it. We'll just dump it into the HTML file. But sure, we can always add later if we want. My only ask before this PR is merged is to add preBody and postBody.

Reply:
OK we'll just dump it then 😆 

Regarding, Gatsby ssr apis. It is kinda different from us in the sense that it allows using React component. I'm not sure how we can replicate that.

We might need `docusaurus-browser.js` or `docusaurus-ssr.js` or something.

This API is much much much lower level as it inserts string directly to the html.

Reply:
So preBodyTags and postBodyTags ?

Reply:
Alright i gotta sleep first.
So

```ts
function injectHtmlTags(): {
  headTags?: HtmlTagObject | HtmlTagObject[];
  preBodyTags?: HtmlTagObject | HtmlTagObject[];
  postBodyTags?: HtmlTagObject | HtmlTagObject[];
};
```

Reply:
>  `preBodyTags` and `postBodyTags`

Sounds good

Reply:
And allowing string ..

Reply:
Yes, woohoo!

Reply:
In this case I wonder if we still need to allow the `stylesheets` and `scripts` in fields in `docusaurus.config.js`.

Reply:
We can deprecate that, but that means user will have to write local plugin ? 😭 

Feel free to send PR if u think its a good idea. Lot of legacy stuff can be pluginified actually. Example: yarn deploy to github pages can be a plugin with `extendCli` instead of core command 😢